demo/src/hooks/useElementSelect.ts

```diff
@@ -1,15 +1,67 @@
-import { useContext } from 'react';
+import { MutableRefObject, useContext, useEffect } from 'react';
 import { SpacesContext } from '../components';
 import { useMembers } from './useMembers';
 
-export const useElementSelect = (element?: string) => {
+import { buildLockId, releaseMyLocks } from '../utils/locking';
+import { Member } from '../utils/types';
+
+export const useElementSelect = (element?: string, lockable: boolean = true) => {
   const space = useContext(SpacesContext);
   const { self } = useMembers();
 
-  const handleSelect = () => {
+  const handleSelect = async () => {
     if (!space || !self) return;
-    space.locations.set({ slide: self.location?.slide, element });
+
+    if (lockable) {
+      const lockId = buildLockId(self.location?.slide, element);
+      const lock = space.locks.get(lockId);
+
+      if (lock?.request.status !== 'locked') {
```

Comment:
`space.locks.get` only ever returns locks in the `LOCKED` state, so this can be:

```suggestion
      if (!lock) {
```